services/app-api/utils/time/time.ts

```diff
@@ -21,18 +21,80 @@ export const convertDateUtcToEt = (date: number): string => {
 };
 
 /*
- * Calculates the period given a due date.
+ * Calculates the period given the current date.
  * The periods are defined as follows:
  *     Period 1 is from 01/01 to 06/30.
  *     Period 2 is from 07/01 to 12/31.
  */
-export const calculatePeriod = (dueDate: number, workPlan?: ReportMetadata) => {
+export const calculatePeriod = (
+  currentDate: number,
+  workPlan?: ReportMetadata
+) => {
   if (workPlan) return workPlan.reportPeriod;
-  const date = new Date(dueDate);
+  const date = new Date(currentDate);
   const period = Math.ceil((date.getMonth() + 1) / 6);
   return period.toString();
 };
 
+/**
+ * Calculates if the given year is a leap year
+ * @param year The year.
+ * @returns if the given year is a leap year.
+ */
+export const isLeapYear = (year: number) => {
+  return (year % 4 == 0 && year % 100 != 0) || year % 400 == 0;
```

Comment:
I think strict equals is probably better here? since the parameter `year` is a `number`? in any case, it doesn't hurt to have the strict comparison here :D